debouncer.go

```diff
@@ -30,9 +31,12 @@ func (d *Debouncer) SendSignal() {
 	defer d.mu.Unlock()
 
 	d.Cancel()
+	d.closeDone()
+	d.done.Store(make(chan struct{}))
```

Comment:
I don't follow this. Isn't that `Cancel()` should unblock any current wait for `d.Done()`?

Reply:
I think it's a bug from the existing code.

The cancel() should clear the old-timer, but it shouldn't unblock d.Done() or reset the channel.
The Done() should be unblocked by a triggered function done()

Is it ok if you clear the `d.done.CompareAndSwap(nil, closedchan)` from Cancel() too.
Maybe I need to test around again after this change to make sure it works as we expected.

Reply:
Hmm, so if we use the same done channel for debouncer, then this code:

```
debouncer.SendSignal()
<-debouncer.Done()

debouncer.UpdateTriggeredFunc(func() {
	*countPtr += 2
})
debouncer.SendSignal()
<-debouncer.Done()
```

won't work as expected, since when the second call to `<-debouncer.Done()` will be unblocked immediately.

Reply:
> Another solution is after we close(done) in the triggered function, we immediately set up a new channel instead of d.done.Store(closedchan), so the next Done() call will get a new one that isn't unblocked immediately.

Seems not work for me, because multiple call to Done will hang. From my understand, each call to `SendSignal` should have separated channels, because it calls `Cancel`.

Reply:
in case (2), if we call Cancel() directly or call through SendSignal(), the channel shouldn't be closed and store `closeChan` to `done`.

The ideal is the Done() is only released by the triggered function instead. Summary just make sure we are the same page:
 - SendSignal() doesn't call closeDone()
 - SendSignal() doesn't update new channel to `d.done`
 - SendSignal() keep to call Cancel()
 - Cancel() doesn't call closeDone()
 - Cancel() doesn't update `closedChan` to `d.done`
 - After TriggeredFunction(), we close old done, setup new channel and store to `d.done`
 - All <-Done() different gorountine will wait until next triggeredFunc() is called.
 - Cancel() and SendSingal() won't close existing <-Done()

Reply:
Hmm, then I think it's not problem at first, because as you explained, multiple call to `Done` will hang.

So I think only the data race need to be fixed.

Reply:
ok, but the idea about closing the channel to broadcast all Done() should be good in my view.
If we want to use it to broadcast data, I think it's good to go if you want.